File: frontend/src/ai/router/AgentRouter.ts
import { ServiceAgent } from '../agents/ServiceAgent';
import { IncidentAgent } from '../agents/IncidentAgent';
import { RCAAgent } from '../agents/RCAAgent';
import { BlastRadiusAgent } from '../agents/BlastRadiusAgent';
import { PredictionAgent } from '../agents/PredictionAgent';
import { WorkflowAgent } from '../agents/WorkflowAgent';
import { ExecutiveAgent } from '../agents/ExecutiveAgent';
import type { AgentConfig, PageType } from '../types';

const agents: Record<PageType, AgentConfig> = {
  service: ServiceAgent,
  incident: IncidentAgent,
  rca: RCAAgent,
  blast: BlastRadiusAgent,
  prediction: PredictionAgent,
  workflow: WorkflowAgent,
  executive: ExecutiveAgent,
};

export function getAgentConfig(pageType: PageType): AgentConfig {
  return agents[pageType] ?? ExecutiveAgent;
}

File: backend/app/services/agent_router.py
"""Route page types to specialized copilot agents."""

from __future__ import annotations

from app.agents import (
    BlastRadiusAgent,
    EarlyDetectionAgent,
    ExecutiveAgent,
    IncidentAgent,
    PredictionAgent,
    RCAAgent,
    ServiceAgent,
    WorkflowAgent,
)
from app.agents.base_agent import BaseAgent

_AGENTS: dict[str, BaseAgent] = {
    "service": ServiceAgent(),
    "incident": IncidentAgent(),
    "rca": RCAAgent(),
    "blast": BlastRadiusAgent(),
    "early_detection": EarlyDetectionAgent(),
    "prediction": PredictionAgent(),
    "workflow": WorkflowAgent(),
    "executive": ExecutiveAgent(),
}


def get_agent(page_type: str) -> BaseAgent:
    return _AGENTS.get(page_type, ExecutiveAgent())
